src/pages/GoalsPage/PartnerGoals.tsx

```diff
@@ -58,7 +58,7 @@ const PartnerGoals = () => {
     const rootGoals = await getRootGoalsOfPartner(relId);
     handleUserGoals(rootGoals);
   };
-
+  refreshActiveGoals();
```

Comment:
this causes infinite re-render.

In useApp, do this 

   addSharedWMGoal(rootGoal)
                  .then(() => {
                    goalWithChildrens.slice(1).forEach((goal) => {
                      addSharedWMGoal(goal).catch((err) => console.log(`Failed to add in inbox ${goal.title}`, err));
                    });
                  })
                  .then(() => {
                    setLastAction("goalNewUpdates");
                  })

Reply:
thanks for your help. 
please take another look and let me know :)